src/java/org/apache/cassandra/cql3/statements/schema/AlterKeyspaceStatement.java

```diff
@@ -78,7 +82,14 @@ public Keyspaces apply(Keyspaces schema)
         validateNoRangeMovements();
         validateTransientReplication(keyspace.createReplicationStrategy(), newKeyspace.createReplicationStrategy());
 
-        return schema.withAddedOrUpdated(newKeyspace);
+        Keyspaces res = schema.withAddedOrUpdated(newKeyspace);
+        if (ClientWarn.instance.getWarnings() != null)
```

Comment:
with proposed earlier method this will be easier:

```
int newSize = ClientWarn.instance.numWarnings();
if (newSize > previousWarningSize)
{
    clientWarnings.addAll(...)
}
```
but even better it can all be probably extraceted into some new `ClientWarn` method,
that would return "getWarningsAfter(previousWarningSize)" and return some list or empty list
depending on the conditions.

Then this code would become simply:
```
clientWarnings.addAll(ClientWarn.instance.getWarningsAfter(previousWarningSize));
```

Reply:
I think this case is not a candidate to be generalized into the `ClientWarn`class. The problem comes from the underlying ThreadLocal and the many execution paths. Some execute on the same thread while others on a pool where that thread is not deterministic. Some will contain previous warnings and others don't. That's why some warnings accumulate and others clear. But that is particular to this specific scenario, hence the `clientWarnings()` on the API to pull warnings across threads. So I wouldn't generalize it.

Reply:
fine